telegram/src/main/kotlin/com/github/kotlintelegrambot/dispatcher/handlers/InputChainHandler.kt

```diff
@@ -0,0 +1,65 @@
+package com.github.kotlintelegrambot.dispatcher.handlers
+
+import com.github.kotlintelegrambot.Bot
+import com.github.kotlintelegrambot.entities.Message
+import com.github.kotlintelegrambot.entities.Update
+import com.github.kotlintelegrambot.extensions.filters.Filter
+
+data class InputChainHandlerEnvironment<T>(
+    val inputChainIdGen: (previousInputChainId: T?, Update) -> T?,
+    val inputChainId: T,
+    val bot: Bot,
+    val update: Update,
+    val message: Message,
+)
+
+class InputChainHandler<T>(
+    private val filter: Filter,
+    private val initialHandler: Handler,
+    private val inputChainIdGen: (previousInputChainId: T?, Update) -> T?,
```

Comment:
How is this supposed to be used? It's not obvious to me by reading the internal code of library, so I think it will be even more difficult for consumers of the library to understand how to (correctly) use this API. Is it possible to encapsulate this detail in the library?

Reply:
I will add some showcases to help other devs to use it and understand how to implement their own chain